apps/src/templates/progress/ProgressLessonTeacherInfo.jsx

```diff
@@ -60,15 +61,21 @@ class ProgressLessonTeacherInfo extends React.Component {
       study: 'hidden-lessons',
       study_group: 'v0',
       event: value,
-      data_json: JSON.stringify({
-        script_name: scriptName,
-        section_id: sectionId,
-        lesson_id: lesson.id,
-        lesson_name: lesson.name
-      })
+      data_json: JSON.stringify(this.firehoseData())
     });
   };
 
+  firehoseData = () => {
+    const {userId, scriptName, section, lesson} = this.props;
+    return {
+      user_id: userId,
```

Comment:
fyi: to pass the user_id, use the `includeUserId` option. https://github.com/code-dot-org/code-dot-org/blob/7e16615a0ee0dfc1105fd6a2a03ee3cf889dd427/apps/src/lib/util/firehose.js#L230

Reply:
👍